src/Routes.tsx

```diff
@@ -14,12 +14,41 @@ import { CommonRoutesFeatures } from './RoutesWrap';
 const { MainView } = lazily(() => import('~views/Main/Main'));
 const { Reports } = lazily(() => import('~views/Reports/Reports'));
 const { ReportPage } = lazily(() => import('~views/Report/Report'));
+const { AboutPage } = lazily(() => import('~views/About/About'));
 const { BivariateManagerPage } = lazily(
   () => import('~views/BivariateManager/BivariateManager'),
 );
 
 export function RoutedApp() {
   const [{ data: userModel, loading }] = useAtom(userResourceAtom);
+
+  function isFirstTimeVisit() {
+    const firstTimeVisit =
+      userModel &&
+      !loading &&
+      location.pathname == '/' &&
+      location.search == '' &&
+      !localStorage.getItem('landed');
+    if (firstTimeVisit) {
+      localStorage.setItem('landed', 'yes');
+      return true;
+    }
+    return false;
+  }
+
+  useEffect(() => {
+    // redirect first-time visitor "/" -> "/about"
+    if (!isFirstTimeVisit()) {
+      return;
+    }
+    const timeoutId = setTimeout(() => {
```

Comment:
why do you need timeout there?

Reply:
if it's done to wait for a userModel loaded, you just need to sync dependencies
this is working without  timeouts

```
  useEffect(() => {
    const isFirstTimeVisit = () => {
      const firstTimeVisit =
        userModel &&
        !loading &&
        location.pathname === '/' &&
        location.search === '' &&
        !localStorage.getItem('landed');
      if (firstTimeVisit) {
        localStorage.setItem('landed', 'true');
        return true;
      }
      return false;
    };

    // redirect first-time visitor "/" -> "/about"
    if (!isFirstTimeVisit()) {
      return;
    }
    history.push(APP_ROUTES.about);
  }, [loading, userModel]);
```

Reply:
the shorted version

  ```
useEffect(() => {
    const isFirstTimeVisit = () =>
      userModel &&
      !loading &&
      location.pathname === '/' &&
      location.search === '' &&
      !localStorage.getItem('landed');

    // redirect first-time visitor "/" -> "/about"
    if (isFirstTimeVisit()) {
      localStorage.setItem('landed', 'true');
      history.push(APP_ROUTES.about);
    }
  }, [loading, userModel]);
```